feg/gateway/services/envoy_controller/control_plane/control_plane.go

```diff
@@ -197,8 +197,6 @@ func GetControllerClient() *ControllerClient {
 	go RunManagementServer(ctx, srv, port)
 	go RunManagementGateway(ctx, srv, gatewayPort)
 
-	<-signal
```

Comment:
what is this channel for? meaning, the cb object is placing something there. What is that

Also note that if the cb object does not place anything there that will be blocked. Maybe it is better to use a timeout to make sure you never block
```
select {
case <-signal:
case <-timeout:
    // log error
}
```